core/src/main/java/org/elasticsearch/http/HttpServer.java

```diff
@@ -71,15 +65,23 @@ protected void doStart() {
         if (logger.isInfoEnabled()) {
             logger.info("{}", transport.boundAddress());
         }
+        registerHttpAddress();
```

Comment:
why is this needed? calling a protected method from the constructor isn't considered a good practice?